src/mds/SessionMap.h

```diff
@@ -666,5 +665,29 @@ class SessionMap : public SessionMapStore {
 
 std::ostream& operator<<(std::ostream &out, const Session &s);
 
+class GatherMsgAcks : public MDSContextSleeper {
+private:
+  std::map<C_Gather *, Session *> unacked;
+
+public:
+  GatherMsgAcks() = delete;
+  GatherMsgAcks(const std::string &name, MDSRank *mds) :
+  MDSContextSleeper(name, mds) {}
+
+  C_Gather *new_sub(Session *session) {
+    unacked.insert(std::pair<C_Gather *, Session *> (sub, session));
+    return new_sub();
+  }
+
+  ContextType *new_sub() override {
```

Comment:
Apparently, `ContextType` is not visible in this file; the code fails to compile. Neither greping over the codebase revealed its definition[1] nor searching on web lead to any clues about it (in case it is defined is standard or boost libraries).

I coud use `MDSInternalContextBase` instead but then these function members would not be overriden.

[1] https://paste.fedoraproject.org/paste/Au5ZsTJghrN9A~c~YkOTYw